x/twap/logic.go

```diff
@@ -259,7 +267,12 @@ func computeTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quote
 }
 
 // twapLog returns the logarithm of the given spot price, base 2.
+// Panics if zero is given.
 func twapLog(price sdk.Dec) sdk.Dec {
+	if price.IsZero() {
+		panic("twap: cannot take logarithm of zero")
+	}
```

Comment:
This LGTM, but I feel like the mental overhead is better here if we make this return an error in such case instead. (We have to be careful that this panic is never called due to this being end block logic. Vs if its an error, we can just check that caller handles errors correctly, which is already the case w/ our errcheck linter)

Reply:
I don't think this panic can ever be reached because we have an additional safeguard in its only caller:
https://github.com/osmosis-labs/osmosis/blob/a15947d8f16c25a750a092c2c82f95bf00dc3732/x/twap/logic.go#L200-L203

This is just a defense-in-depth sanity check.

The problem with returning an error is that it would require a large refactor to propagate this error to multiple callers of `recordWithUpdatedAccumulators`.

I'm going to keep this as is and merge it. However, please let me know if you have further concerns, happy to discuss and address in another PR if needed